File: 3804. Number of Centered Subarrays/Solution.cpp
#include <unordered_set>
#include <vector>
using namespace std;

class Solution {
public:
    int centeredSubarrays(vector<int>& nums)
    {
        unordered_set<int> seen;
        int n = nums.size(), result = 0;
        for (int i = 0; i < n; ++i) {
            seen.clear();
            int s = 0;
            for (int j = i; j < n; ++j) {
                s += nums[j];
                seen.insert(nums[j]);
                result += seen.contains(s);
            }
        }
        return result;
    }
};

File: 3804. Number of Centered Subarrays/Solution.py
class Solution:
    def centeredSubarrays(self, nums: list[int]) -> int:
        n = len(nums)
        result = 0
        for i in range(n):
            seen = set()
            s = 0
            for j in range(i, n):
                seen.add(nums[j])
                s += nums[j]
                if s in seen:
                    result += 1
        return result
